buildSrc/src/test/java/org/opensearch/gradle/DistributionDownloadPluginTests.java

```diff
@@ -79,6 +80,54 @@ public void testVersionDefault() {
         assertEquals(distro.getVersion(), VersionProperties.getOpenSearch());
     }
 
+    public void testCustomDistributionUrlWithUrl() {
+        Project project = createProject(null, false);
+        String customUrl = "https://artifacts.opensearch.org/custom";
+        project.getExtensions().getExtraProperties().set("customDistributionUrl", customUrl);
+        DistributionDownloadPlugin plugin = project.getPlugins().getPlugin(DistributionDownloadPlugin.class);
+        plugin.setupDistributions(project);
+        assertEquals(4, project.getRepositories().size());
+        assertEquals(
+            ((DefaultIvyArtifactRepository) project.getRepositories().getAt("opensearch-downloads")).getUrl().toString(),
+            customUrl
+        );
+        assertEquals(
+            ((DefaultIvyArtifactRepository) project.getRepositories().getAt("opensearch-snapshots")).getUrl().toString(),
+            customUrl
+        );
+        assertEquals(
+            ((DefaultIvyArtifactRepository) project.getRepositories().getAt("elasticsearch-downloads")).getUrl().toString(),
+            "https://artifacts-no-kpi.elastic.co"
+        );
+        assertEquals(
+            ((DefaultIvyArtifactRepository) project.getRepositories().getAt("elasticsearch-snapshots")).getUrl().toString(),
+            "https://snapshots-no-kpi.elastic.co"
+        );
+    }
+
+    public void testCustomDistributionUrlWithoutUrl() {
+        Project project = createProject(null, false);
+        DistributionDownloadPlugin plugin = project.getPlugins().getPlugin(DistributionDownloadPlugin.class);
+        plugin.setupDistributions(project);
+        assertEquals(5, project.getRepositories().size());
```

Comment:
Why do we still get 5 here? (And check for 4 below)

Reply:
Yes URL will be same both times. But I think either if we want to remove the second repo from repositories set or add a test check(opensearch-downloads2) again for same URL, we need to perform extra operation. Please correct me if I am wrong, adjusting the test to check all values would be like an extra safe step and better than removing repo. So, adding test (checking opensearch-downloads2) to check all values as you mentioned in your previous suggestion.

Reply:
works for me.